### PR TITLE
Remove Optimism estimator validation

### PR DESCRIPTION
**Pull request by dimriou**
This removes Optimism's constraint on estimator type so we can test the new dynamic gas price model.

### CODE REVIEW

File: core/chains/evm/client/errors.go [hunk 1]
```diff
@@ -128,10 +128,9 @@ var arbitrum = ClientErrors{
 	L2Full:                regexp.MustCompile(`(: |^)(queue full|sequencer pending tx pool full, please try again)(:|$)`),
 }
 
-var optimism = ClientErrors{
-	L2FeeTooLow:  regexp.MustCompile(`(: |^)fee too low: \d+, use at least tx.gasLimit = \d+ and tx.gasPrice = \d+$`),
-	L2FeeTooHigh: regexp.MustCompile(`(: |^)fee too high: \d+, use less than \d+ \* [0-9\.]+$`),
-}
+// Optimism Bedrock introduced the same errors as geth
+// https://github.com/ethereum-optimism/op-geth/blob/optimism/core/error.go
+var optimism = geth
```
Comment (samsondav) [inline core/chains/evm/client/errors.go:134]: this is gonna need to stay and be marked for removal later to support both versions at once

File: core/chains/evm/config/v2/defaults/Optimism_Alpha.toml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+LinkContractAddress = '0xdc2CC710e42857672E7907CF474a69B63B93089f'
+
+NoNewHeadsThreshold = '60s'
+TxResendAfterThreshold = '30s'
+
+[GasEstimator]
+EIP1559DynamicFees = true
+
+[GasEstimator.BlockHistory]
+BlockHistorySize = 24
```
Comment (samsondav) [inline core/chains/evm/config/v2/defaults/Optimism_Alpha.toml:13]: Why this value?

File: core/chains/evm/config/v2/defaults/Optimism_Alpha.toml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+ChainID = '28528'
+ChainType = 'optimism'
+FinalityDepth = 200
+LinkContractAddress = '0xdc2CC710e42857672E7907CF474a69B63B93089f'
+
+NoNewHeadsThreshold = '60s'
```
Comment (samsondav) [inline core/chains/evm/config/v2/defaults/Optimism_Alpha.toml:6]: Does optimism bedrock regularly emit heads?

File: core/chains/evm/config/v2/defaults/Optimism_Alpha.toml (hunk 2 above)
Comment (dimriou) [reply core/chains/evm/config/v2/defaults/Optimism_Alpha.toml:13]: It was an attempt to bypass the issue with the estimator and the unrecognized tx type we discussed. We can change the value after our tests.

**Pull request closed without merging**